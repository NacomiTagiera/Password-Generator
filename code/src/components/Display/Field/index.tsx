interface Props {
  password: string;
}

export default function Field({ password }: Props) {
  return <input type="password" value={password} />;
}
