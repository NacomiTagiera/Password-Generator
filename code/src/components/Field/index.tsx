import { TextField } from "@mui/material";

interface Props {
  password: string;
}

export default function Field({ password }: Props) {
  return (
    <TextField
      fullWidth
      value={password}
      variant="standard"
      placeholder="Your Password"
      InputProps={{
        "aria-readonly": "true",
        readOnly: true,
      }}
    />
  );
}
