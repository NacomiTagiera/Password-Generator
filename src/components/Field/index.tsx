import { TextField } from '@mui/material';

export const Field = ({ password }: { password: string }) => {
  return (
    <TextField
      name='password'
      fullWidth
      value={password}
      variant='standard'
      placeholder='Your Password'
      InputProps={{
        readOnly: true,
      }}
    />
  );
};
