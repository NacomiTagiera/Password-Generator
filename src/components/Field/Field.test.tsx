import { render, screen } from '@testing-library/react';

import Field from '.';

describe('Field', () => {
  const password = '1234567890';

  beforeEach(() => {
    render(<Field password={password} />);
  });

  it('renders password value correctly', () => {
    const fieldElement = screen.getByPlaceholderText('Your Password');
    expect(fieldElement).toHaveValue(password);
  });

  it('displays correct label', () => {
    const fieldLabel = screen.getByPlaceholderText('Your Password');
    expect(fieldLabel).toBeInTheDocument();
  });
});
