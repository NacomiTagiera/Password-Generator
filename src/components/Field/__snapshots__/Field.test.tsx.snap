// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    data-testid="password-field"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-953pxc-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input Mui-readOnly MuiInputBase-readOnly css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r0:"
        name="password"
        placeholder="Your Password"
        readonly=""
        type="text"
        value="123"
      />
    </div>
  </div>
</DocumentFragment>
`;
